From: Mathieu Baudier <mbaudier@argeo.org>
Date: Tue, 11 Mar 2025 15:40:11 +0100
Subject: use-llama-cpp-library

---
 examples/talk-llama/CMakeLists.txt | 25 ++++---------------------
 1 file changed, 4 insertions(+), 21 deletions(-)

diff --git a/examples/talk-llama/CMakeLists.txt b/examples/talk-llama/CMakeLists.txt
index aea1ae6..f86a338 100644
--- a/examples/talk-llama/CMakeLists.txt
+++ b/examples/talk-llama/CMakeLists.txt
@@ -3,29 +3,12 @@ if (WHISPER_SDL2)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
     set(TARGET whisper-talk-llama)
-    add_executable(${TARGET} talk-llama.cpp
-        llama.cpp
-        llama-adapter.cpp
-        llama-arch.cpp
-        llama-batch.cpp
-        llama-chat.cpp
-        llama-context.cpp
-        llama-cparams.cpp
-        llama-grammar.cpp
-        llama-hparams.cpp
-        llama-impl.cpp
-        llama-kv-cache.cpp
-        llama-mmap.cpp
-        llama-model-loader.cpp
-        llama-model.cpp
-        llama-quant.cpp
-        llama-sampling.cpp
-        llama-vocab.cpp
-        unicode.cpp
-        unicode-data.cpp)
+    add_executable(${TARGET} talk-llama.cpp)
     target_include_directories(${TARGET} PRIVATE ${SDL2_INCLUDE_DIRS})
 
-    target_link_libraries(${TARGET} PRIVATE common common-sdl whisper ${SDL2_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+    target_link_libraries(${TARGET} PRIVATE common common-sdl whisper llama ${SDL2_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+
+	install(TARGETS ${TARGET} RUNTIME)
 
     if(WIN32)
         # It requires Windows 8.1 or later for PrefetchVirtualMemory
